src/waveform/renderers/waveformbeat.cpp

```diff
@@ -74,3 +74,19 @@ QPolygonF WaveformBeat::getEquilateralTriangle(float edgeLength,
     polygon << firstPoint << secondPoint << thirdPoint;
     return polygon;
 }
+
+bool WaveformBeat::contains(QPoint point, Qt::Orientation orientation) const {
+    Q_UNUSED(orientation);
+    return (m_orientation == Qt::Horizontal &&
+                   (m_position - kClickableLinePaddingPixels < point.x() &&
+                           point.x() <
+                                   m_position + kClickableLinePaddingPixels)) ||
+            (m_orientation == Qt::Vertical &&
+                    (m_position - kClickableLinePaddingPixels < point.y() &&
+                            point.y() <
+                                    m_position + kClickableLinePaddingPixels));
```

Comment:
These heavily nested conditional statements are hard to read. Can you check the orientation in an if/else block?

Reply:
And and maybe use a temp var to do the kClickableLinePaddingPixels addition/subtraction only once.